23_Arithmetic2UnitTest/app_resources/testWorkgroup.comp.hlsl

```diff
@@ -1,20 +1,64 @@
 #pragma shader_stage(compute)
 
-#include "workgroupCommon.hlsl"
+#include "nbl/builtin/hlsl/glsl_compat/core.hlsl"
+#include "nbl/builtin/hlsl/glsl_compat/subgroup_basic.hlsl"
+#include "nbl/builtin/hlsl/subgroup2/arithmetic_portability.hlsl"
+#include "nbl/builtin/hlsl/workgroup2/arithmetic.hlsl"
+
+static const uint32_t WORKGROUP_SIZE = 1u << WORKGROUP_SIZE_LOG2;
+
+#include "shaderCommon.hlsl"
+
+using config_t = nbl::hlsl::workgroup2::ArithmeticConfiguration<WORKGROUP_SIZE_LOG2, SUBGROUP_SIZE_LOG2, ITEMS_PER_INVOCATION>;
+
+typedef vector<uint32_t, config_t::ItemsPerInvocation_0> type_t;
+
+// final (level 1/2) scan needs to fit in one subgroup exactly
+groupshared uint32_t scratch[config_t::ElementCount];
```

Comment:
beware, if the variable is not optimized out (either due to lack of constant folding for the branch where its used) or due to optimizations being disabled https://godbolt.org/z/rbMeP4jfT

You get a compile error.

Its also illegal in C++ https://stackoverflow.com/questions/9722632/what-happens-if-i-define-a-0-size-array-in-c-c